### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by binyamolango**
# Hi Reviewer 👋😊

> Here are the changes that we implemented to the project

- Created React App.
- Installed React Redux, Redux Logger, and React Router.
- Configured Linter.
- Downloaded the free planet image for the app logo.
- Created routes and view components for rockets, missions, dragons, and my profile.
- Used <NavLink /> for the page navigation links and style active class to indicate which section/page the user is currently on.
- Create directories for all Redux state slice files rockets, missions, and dragons.
- Fetched data from the SpaceX API endpoints.
- Dispatched an action to store the selected data in the Redux store.
- Used useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes.
- Styled the different sections by hand and using react-bootstrap.
- Rendered a list of rockets, a table with the missions' data, and a list of dragons (as per design).
- Wrote actions and reducers for booking rockets/dragons and joining missions.
- Wrote actions and reducers for canceling rockets/dragons and leaving missions.
- Rendered a list of all joined missions, a list of all reserved rockets, and a list of all reserved dragons (use filter()).
- Set up testing libraries(React Testing Library and Jest).

### CODE REVIEW

File: src/redux/rocket/rocketSlice.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import axios from 'axios';
```
Comment (omar-muhamad) [inline src/redux/rocket/rocketSlice.js]: - [ ] Please remove this eslint disablement line and fix linters issues yourself. Linters is a tool to help you write clean and error-free code so please don't disable it and try to fix it's errors.

File: src/redux/mission/MissionsSlice.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import axios from 'axios';
```
Comment (omar-muhamad) [inline src/redux/mission/MissionsSlice.js]: - [ ] same here, Please remove this eslint disablement line and fix linters issues yourself. Linters is a tool to help you write clean and error-free code so please don't disable it and try to fix it's errors.

File: src/component/Dragon/DragonList.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import './dragon.css';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import PropTypes from 'prop-types';
```
Comment (omar-muhamad) [inline src/component/Dragon/DragonList.js]: - [ ] Same here, Please remove this eslint disablement line and fix linters issues yourself. Linters is a tool to help you write clean and error-free code so please don't disable it and try to fix it's errors.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi Team,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @yusufyusie, @MayPyone  & @binyamolango 

Good job implementing the suggested changes💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Styling is looking great
-  ✅ Implemented suggestions 
-  ✅ Your README looks good
-  ✅ Linters are working properly

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**